### PR TITLE
QA Test for New Marketing Component

### PR DESCRIPTION
**Pull request by jlieb10**
## Why are you doing this?

We have a design for a new marketing component and this needs to be 'QA tested' alongside our current marketing component. We aren't looking for anything scientific in this test, just want to watch the two alongside each other for a couple days to make sure we are getting opt-ins through both. Then we will be able to switch over to the 'newer' component (with checkbox).

Doing this as an A/B test was unexpectedly complicated and involved touching a lot of files that I didn't expect to need to touch. I have added inline documentation to help with removing this as a test when that time comes.

[**Trello Card**](https://trello.com)

## Changes

* Created new marketing option as `MarketingComponentWithCheckbox`
* Implemented it across contributions thank you pages as an AB test against the current `MarketingComponent`

## Screenshots
Control:
marketing abtest control

Variant ("new"):
marketing abtest variant

### CODE REVIEW

File: support-frontend/assets/components/marketingConsent/marketingConsent.jsx [hunk 1]
```diff
@@ -13,19 +13,26 @@ import { logException } from 'helpers/logger';
...
   email: string,
   csrf: CsrfState,
-  onClick: (?string, CsrfState) => void,
+  onClick: (?string, CsrfState, ?ThankYouPageMarketingComponentTestVariants) => void,
   requestPending: boolean,
+  marketingComponentVariant?: ThankYouPageMarketingComponentTestVariants,
 |};
 
 type PropTypes = {|
-  ...ButtonPropTypes,
```
Comment (jlieb10) [inline support-frontend/assets/components/marketingConsent/marketingConsent.jsx]: For some reason, doing the object spread here was causing eslint errors - I don't think it likes it when it comes to optional props

**Comment by tomrf1** [comment]
I'm not sure I understand the reason for doing this as an a/b test. Surely we can tell if the new component is working or not regardless?

File: support-frontend/assets/components/marketingConsent/marketingConsent.jsx (hunk 1 above)
Comment (tomrf1) [reply support-frontend/assets/components/marketingConsent/marketingConsent.jsx]: that's annoying. At least we can revert this part later

Comment (jlieb10) [reply support-frontend/assets/components/marketingConsent/marketingConsent.jsx]: 👍

**Pull request merged**